app/src/views/private/components/drawer-item.vue

```diff
@@ -177,8 +177,11 @@ export default defineComponent({
 
 		const fields = computed(() => {
 			if (props.circularField) {
-				return fieldsWithPermissions.value.filter((field: Field) => {
-					return field.field !== props.circularField;
+				return fieldsWithPermissions.value.map((field: Field) => {
+					if (field.field === props.circularField) {
+						field.meta!.readonly = true;
```

Comment:
I think there's a tiny change `meta` is `null` (when it's a database-only field with no directus_fields record). Lets use `set` from lodash to prevent an error when `meta` doesn't exist.

```suggestion
						set(field, 'meta.readonly', true);
```